### ISSUE TITLE
Drag and drop in Group's Tree (in file bib/ shared database)

### ISSUE THREAD

**Issue by robyquin** [JabRef/jabref#9368]
### JabRef version

5.7 (latest release)

### Operating system

Windows

### Details on version and operating system

Windows 10

### Checked with the latest development build

- [X] I made a backup of my libraries before testing the latest development version.
- [X] I have tested the latest development version and the problem persists

### Steps to reproduce the behaviour

When I drag and drop a group to an "Automatic * Group" destination, the moved group is no longer displayed in the tree.

This happens because the group becomes the child of an "Automatic Group", which does not allow having children outside the collection criteria

This bug also happens in the shared database, but drag and drop is a changement that is not synchronized, and therefore less dangerous.

### Appendix

> @Comment{jabref-meta: databaseType:biblatex;}

> @Comment{jabref-meta: grouping:
> 0 AllEntriesGroup:;
> 1 AutomaticKeywordGroup:By Author\;0\;author\;,\;>\;0\;0x800080ff\;ACCOUNT\;\;;
> 1 AutomaticKeywordGroup:By CrossReference\;2\;crossref\;,\;>\;0\;0x800080ff\;FOLDER_REFRESH_OUTLINE\;\;;
> 1 AutomaticKeywordGroup:By Journal\;0\;journal\;,\;>\;0\;0x800080ff\;NOTEBOOK_OUTLINE\;\;;
> 1 AutomaticKeywordGroup:By Gruops\;0\;groups\;,\;>\;0\;0x800080ff\;FOLDER_EDIT_OUTLINE\;\;;
> 1 AutomaticKeywordGroup:By Organization\;0\;organization\;,\;>\;0\;0x800080ff\;ACCOUNT_SUPERVISOR\;\;;
> 1 AutomaticKeywordGroup:By KeyWords\;0\;keywords\;,\;>\;0\;0x800080ff\;KEY_VARIANT\;\;;
> **2 StaticGroup:WebSite\;0\;1\;0xff9966ff\;\;\;;**
> 1 StaticGroup:Categories\;0\;1\;0x800080ff\;FOLDER_MULTIPLE_PLUS\;\;;
> 2 StaticGroup:Developement Software\;0\;1\;0x4d66ccff\;\;\;;
> }

**Comment by robyquin** [comment]
With branches [fix-issue-9368](https://github.com/robyquin/jabref/tree/fix-issue-9368) on my fork, I tried to study a solution to this issue.

Briefly, the solution is to add two methods to the various types of groups: **AllowDragInto**, **CanBeDragged**.

During drag and drop, the GroupTreeView queries these methods to know what is allowed to do.

If the drag and drop is successful, in shared database, update `grouping` in the MetaData table.

**NOTE**: Working on this patch I noticed that the actions allowed by the context menus of the GroupTreeView are not all necessary, I could fix this as well if another problem didn't open.
